dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.dependencies;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Aether initialization.
+ */
+class RepositoryUtility {
+
+  static RepositorySystem newRepositorySystem() {
+    DefaultServiceLocator locator = MavenRepositorySystemUtils.newServiceLocator();
+    locator.addService(RepositoryConnectorFactory.class, BasicRepositoryConnectorFactory.class);
+    locator.addService(TransporterFactory.class, FileTransporterFactory.class);
+    locator.addService(TransporterFactory.class, HttpTransporterFactory.class);
+  
+    return locator.getService(RepositorySystem.class);
+  }
+
+  static RepositorySystemSession newSession(RepositorySystem system ) {
+    DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+  
+    LocalRepository localRepository = new LocalRepository(findLocalRepository().getAbsolutePath());
+    session.setLocalRepositoryManager(system.newLocalRepositoryManager(session, localRepository));
+    session.setReadOnly();
+    return session;
+  }
+
+  @VisibleForTesting
+  static File findLocalRepository() {
+    Path home = Paths.get(System.getProperty("user.home"));
```

Comment:
Is this dangerous, in case someone has a corrupt local repository?

Is the advantage here because of cache across runs, or mainly because of cache within a run? If the latter, can we just use a temp directory?

Reply:
Depends on what you mean by dangerous. This isn't safety critical. It can fail, but most Maven devs have had to nuke their repository and start over at some point. It's unlikely we'll have a local Maven repo that i corrupt in such a way that the build gives incorrect data instead of simply erroring out. We may need to reconsider if we support snapshots. 

I believe the advantage is both across runs and within a  run.